### ISSUE TITLE
`iamax` has no device implementation

### ISSUE THREAD

**Issue by PTNobel** [icl-utk-edu/blaspp#45]
I am trying to implement an algorithm that uses `iamax` on a GPU, but there is no support right now for this.

Would it be possible to add a version of `iamax` that takes a queue argument and runs on GPU memory? I think it could be backed on CUDA by the `amax` function.

**Comment by mgates3** [comment]
Yes, this should be feasible, but will take a little while to get into our work queue.
